src/domain/bitcoin/lightning/index.types.d.ts

```diff
@@ -3,23 +3,12 @@ type LnInvoiceDecodeError = import("./errors").LnInvoiceDecodeError
 type LightningServiceError = import("./errors").LightningServiceError
 type RouteNotFoundError = import("./errors").RouteNotFoundError
 
-declare const invoiceExpirationSymbol: unique symbol
-type InvoiceExpiration = Date & { [invoiceExpirationSymbol]: never }
-
-declare const encodedPaymentRequestSymbol: unique symbol
-type EncodedPaymentRequest = string & { [encodedPaymentRequestSymbol]: never }
-
-declare const paymentHashSymbol: unique symbol
-type PaymentHash = string & { [paymentHashSymbol]: never }
-
-declare const paymentSecretSymbol: unique symbol
-type PaymentSecret = string & { [paymentSecretSymbol]: never }
-
-declare const featureBitSymbol: unique symbol
-type FeatureBit = number & { [featureBitSymbol]: never }
-
-declare const featureTypeSymbol: unique symbol
-type FeatureType = string & { [featureTypeSymbol]: never }
+type InvoiceExpiration = Date & { readonly brand: unique symbol }
+type EncodedPaymentRequest = string & { readonly brand: unique symbol }
+type PaymentHash = string & { readonly brand: unique symbol }
+type PaymentSecret = string & { readonly brand: unique symbol }
+type FeatureBit = number & { readonly brand: unique symbol }
+type FeatureType = string & { readonly brand: unique symbol }
```

Comment:
As far as I can tell it gets us what we want from the type-checker

![image](https://user-images.githubusercontent.com/17693119/147359559-4651ad4d-da6c-47dd-b54d-2baa598b88cc.png)

```suggestion
type Unique = { readonly brand: unique symbol }

type InvoiceExpiration = Date & Unique
type EncodedPaymentRequest = string & Unique
type PaymentHash = string & Unique
type PaymentSecret = string & Unique
type FeatureBit = number & Unique
type FeatureType = string & Unique
```

Reply:
If we like this suggestion, this find & replace should work from VSCode

![image](https://user-images.githubusercontent.com/17693119/147361096-6ff5c348-25b3-4dc9-9321-ea9db56cbe4b.png)

Reply:
you master better VSCode than I do :) do you want to run it then on a new PR? we can close this one afterwards

Reply:
Sure can do

Reply:
can you double check that with the new type `Unique` you created that 2 types of `string & Unique` are not interchangeables (as opposed to a `Date` and `Date & Unique`)

Reply:
Ok I found an issue with this approach in general.

I can do `const userId: UserId = 'id' as WalletId` and this works because it's comparing `{readonly brand: unique symbol}` in both cases. This isn't a problem with our current approach because we also define unique symbol props instead of the generic `brand` prop

Reply:
what would be the message if we were to do `const userId: UserId = 'id' as WalletId` with our approach?

Reply:
No error, that's seen as valid code

Reply:
I just did a test, the issue seems to be using `type Unique = { readonly brand: unique symbol }`. otherwise just doing:

```
type PaymentHash = string & { readonly brand: unique symbol }
type PaymentSecret = string & { readonly brand: unique symbol }
``` 

seems to work

Reply:
![image](https://user-images.githubusercontent.com/11156370/147404277-827bf532-9bcf-47e0-b4c6-926f6668e67f.png)

![image](https://user-images.githubusercontent.com/11156370/147404271-3a7e40ea-a6ce-4a1b-bf56-34f779cd6111.png)